### ISSUE TITLE
Magically associate inputs with controls to force outcome

### ISSUE THREAD

**Issue by clintoncwolfe** [inspec/inspec#3845]
### 🎛 Description

Refs #3017 

One idea that comes up occasionally (see also the secondary topic on #3840) is a desire to alter the output of a control based on an input (formerly known as attributes). Today, you would need to have something that has a naming standard for inputs, explicitly examines the loaded controls, then call something like `skip_control` on each that was marked to skip or fail (or, from #3017, have a finer-grained outcome, like 'waivered' or 'exception' or 'POAM').

If we adopted a naming standard for the inputs, InSpec itself could handle the latter two parts.

### 🌍 InSpec and Platform Version

3.6.x

### 💁 Possible Solutions

Post ideas below for naming conventions.

**Comment by clintoncwolfe** [comment]
Attn @TheLunaticScripter
Attn @mschmeitz

**Comment by aaronlippold** [comment]
I like this idea: the one thing to keep in mind is that the above classes waiver, plan of action and milestones (POAM) and exception are all a reporting decisions that are separate from the test results themselves and for the most part we accreditors still want and should see the results of the tests by default regardless of if we decide in our assessment that the results of those tests do or do not have impact on the final evaluation.

This mainly says that we should not have the tests skipped by default - run the tests and show the result. In our reporter however we collect them in the 'Non-Applicable' group ( the general term for the above classes ) for reporting and the reporting impact should be none or 0. Lastly, I think in this class we could add an opt to say skip the tests but that must be a user set field and - I stress - not the default.
The example: We did an assessment on a system and for the test event and the Approving Official selected a set of tests that would be Non-Applicable. We provided a dependent profile that updated the desc of those controls to include agree additional verbiage of why they were Non-Applicable for reporting. The approving official agreed that these controls did not need to be 'assessed' - count against the final evaluation - however that they still wanted to know the state of those controls (aka the test results either good or bad) so that they could have a clear understanding of the relative risk of the system.
